### PR TITLE
feat(daemon): introduce reboot cleanup function

### PR DESCRIPTION
**Pull request by anpep**

### CODE REVIEW

**Comment by flotter** [comment]
This PR adds another low level restart handler without changing the internal restart logic (i.e. restart manager) of the Daemon.

In a previous change, we (with the approval of CTO) added the syscall reboot handler. However, further work on a derivative system is proving that reboot request invocation inside the daemon is too limiting for our use-case. It either requires system knowledge passed into the daemon for cleanup, or the daemon can stay mostly system agnostic, but request cleanup + reboot to be dealt with by the caller.

```
1. Process entry point
2. Process system bootstrap
3. Daemon/Overlord start
:
4. Daemon/Overlord stop
5. Process system un-bootstrap (sync / umount / etc...)
6. Issue reboot (if requested)
```

File: internals/daemon/daemon.go [hunk 1]
```diff
@@ -773,6 +778,13 @@ func syscallModeReboot(rebootDelay time.Duration) error {
 	return nil
 }
 
+func externalModeReboot(rebootDelay time.Duration) error {
+	if rebootDelay > 0 {
+		time.Sleep(rebootDelay)
+	}
```
Comment (pedronis) [inline internals/daemon/daemon.go:784]: doing this sleep inside HandleRestart seems wrong, you might have to distinguish the calls from HandleRestart vs doReboot now

**Comment by flotter** [comment]
@anpep 

Regarding the timeout comment - inside ```HandleRestart``` there is a "backup" reboot scheduled for in-case the daemon teardown gets stuck. So as per Samuele's comment, that would not make sense to also defer to an external reboot. Let's brainstorm this through again.

Ben was asking if we should not just bypass the daemon here (which is what we have done). So I think we just revisit this to make sure out approach makes sense.

**Comment by anpep** [comment]
Closing for now, since it clearly needs more thought than I put into it initially. Let's work on this during the next cycle! Thanks for your feedback

**Pull request closed without merging**